src/transformers/models/splinter/modeling_splinter.py

```diff
@@ -0,0 +1,1556 @@
+        return self.splinter_qass if not self.initialize_new_qass else self.new_splinter_qass
+
+    @add_start_docstrings_to_model_forward(SPLINTER_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
+    @add_code_sample_docstrings(
+        tokenizer_class=_TOKENIZER_FOR_DOC,
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=QuestionAnsweringModelOutput,
+        config_class=_CONFIG_FOR_DOC,
+    )
+    def forward(
+        self,
+        input_ids=None,
+        attention_mask=None,
+        token_type_ids=None,
+        position_ids=None,
+        head_mask=None,
+        inputs_embeds=None,
+        start_positions=None,
+        end_positions=None,
+        output_attentions=None,
+        output_hidden_states=None,
+        return_dict=None,
+        question_positions=None,
+    ):
+        r"""
+        start_positions (:obj:`torch.LongTensor` of shape :obj:`(batch_size,)`, `optional`):
+            Labels for position (index) of the start of the labelled span for computing the token classification loss.
+            Positions are clamped to the length of the sequence (:obj:`sequence_length`). Position outside of the
+            sequence are not taken into account for computing the loss.
+        end_positions (:obj:`torch.LongTensor` of shape :obj:`(batch_size,)`, `optional`):
+            Labels for position (index) of the end of the labelled span for computing the token classification loss.
+            Positions are clamped to the length of the sequence (:obj:`sequence_length`). Position outside of the
+            sequence are not taken into account for computing the loss.
+        question_positions (:obj:`torch.LongTensor` of shape :obj:`(batch_size, num_questions)`, `optional`):
+            The positions of all question tokens. If given, start_logits and end_logits will be of shape
+            :obj:`(batch_size, num_questions, sequence_length)`. If None, the first question token in each sequence in
+            the batch will be the only one for which start_logits and end_logits are calculated and they will be of
+            shape :obj:`(batch_size, sequence_length)`.
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        question_positions_were_none = False
+        if question_positions is None:
+            if input_ids is not None:
+                question_position_for_each_example = torch.argmax(
+                    (torch.eq(input_ids, self.question_token_id)).int(), dim=-1
+                )
+            else:
+                question_position_for_each_example = torch.zeros(
+                    inputs_embeds.size(0), dtype=torch.int64, layout=inputs_embeds.layout, device=inputs_embeds.device
+                )
+            question_positions = question_position_for_each_example.unsqueeze(-1)
+            question_positions_were_none = True
+
+        if attention_mask is not None:
+            attention_mask[input_ids == 102] = 0
```

Comment:
This shouldn't be hardcoded here I think - what token is 102?

Reply:
Removed this part, it's not crucial